cloudinit/analyze/show.py

```diff
@@ -243,19 +242,6 @@ def gather_timestamps_using_systemd():
         # lxc based containers do not set their monotonic zero point to be when
         # the container starts, instead keep using host boot as zero point
         if util.is_container():
```

Comment:
I don't know many people using cloud-init in jails, but i need to test if this code works in FreeBSD

Reply:
@igalic +1 good idea, upstream cloud-init gets lots of usage and testing in lxc, but it would be good to know how this code performs in jails. Just running the following in a jail would probably tell us all that we need to know:
```python3
#!/usr/bin/python3
import os, time
from cloudinit.util import uptime
print(f"{uptime(),os.stat('/proc/1/cmdline').st_atime, time.monotonic()}")
```

Reply:
well, there is no '/proc/1` in a FreeBSD jail, and if there was, there'd be no `/proc/1/cmdline`

FreeBSD doesn't provide PID name spaces (yet, like Solaris zones), and the `/proc` is very stripped down for security reasons

Reply:
> well, there is no '/proc/1`in a FreeBSD jail, and if there was, there'd be no`/proc/1/cmdline`
> 
> FreeBSD doesn't provide PID name spaces (yet, like Solaris zones), and the `/proc` is very stripped down for security reasons

I see - the others would still be useful to know

Reply:
`uptime()` takes a different path on BSD, so no `/proc/1` needed.

Reply:
for completeness:

```python
>>> print(f"{uptime(), time.monotonic()}")
('4598464.319990158', 4598464.320018191)
>>> os.system("uptime")
 9:20PM  up 53 days,  5:24, 0 users, load averages: 0.16, 0.23, 0.20
```

by default, FreeBSD jails don't have `/proc` mounted; uptime is a syscall which returns the kernel's uptime.

```
root@ports:~ # ps awuxdj
USER   PID %CPU %MEM   VSZ  RSS TT  STAT STARTED    TIME COMMAND           PPID  PGID   SID JOBC
root 21923  0.0  0.0 12768 2016  -  SsJ  Mon16   0:00.52 /usr/sbin/syslog     1 21923 21923    0
root 31780  0.0  0.1 14000 3324  5  SJ   21:10   0:00.06 tcsh             31429 31780 31429    1
root 12419  0.0  0.1 13600 2580  5  R+J  21:27   0:00.00 ps awuxdj        31780 12419 31429    1
root 50023  0.0  0.1 16560 2108  3  I+J  Mon16   0:00.03 tcsh             49556 50023 49556    1
root@ports:~ # 
```

the oldest process' PPID here is the host system's PID 1

Anyway, I'll look into this as soon as people start seriously using cloud-init in jails, or if I find a good way of testing it under jails

Reply:
@igalic thanks! it looks like `uptime()` should work for jails too

Reply:
uptime works, but it's the host's uptime, so often can be a lie. then again, I don't know how important this is to be accurate

Reply:
+1 Sounds good. This isn't a very high priority without a bunch of users. This change is only relevant for `analyze` subcommands - users that are manually digging into performance.